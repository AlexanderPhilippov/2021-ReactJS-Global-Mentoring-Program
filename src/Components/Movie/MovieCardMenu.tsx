import React, { useState } from 'react'
import { MovieActions } from './models'

const MovieCardMenu: React.FC<MovieActions> = ({
    handleEdit,
    handleDelete,
}) => {
    const [isOpen, setIsOpen] = useState(false)

    return !isOpen ? (
        <div
            className="movie-card__menu_closed"
            onClick={() => setIsOpen(true)}
        ></div>
    ) : (
        <div className="movie-card__menu_opened">
            <div onClick={() =>{handleEdit(); setIsOpen(false)}}>Edit</div>
            <div onClick={() => {handleDelete(); setIsOpen(false)}}>Delete</div>
            <div onClick={() => setIsOpen(false)}>Close menu</div>
        </div>
    )
}

export default MovieCardMenu
