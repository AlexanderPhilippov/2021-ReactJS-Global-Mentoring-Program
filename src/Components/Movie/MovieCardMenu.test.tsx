/**
 * @jest-environment jsdom
 */

import React from 'react'
import MovieCardMenu from './MovieCardMenu'
import { render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()
const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useDispatch: () => mockDispatch,
}))

const setIsOpenMock = jest.fn()
jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: () => [true, setIsOpenMock],
}))

it('MovieCardMenu Opened', () => {
    const store = mockStore({})
    const { asFragment, debug, getByText } = render(
        <Provider store={store}>
            <MovieCardMenu movieId={1234} />
        </Provider>
    )
    console.log(debug())
    fireEvent.click(getByText(/edit/i))
    fireEvent.click(getByText(/delete/i))
    fireEvent.click(getByText(/close menu/i))
    console.log(debug())
    expect(mockDispatch).toBeCalledTimes(2)
    expect(setIsOpenMock).toBeCalledTimes(3)
    expect(asFragment()).toMatchSnapshot()
})
