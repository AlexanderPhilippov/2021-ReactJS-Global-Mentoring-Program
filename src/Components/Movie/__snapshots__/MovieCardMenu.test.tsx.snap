// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardMenu Opened 1`] = `
<DocumentFragment>
  <div
    class="movie-card__menu_opened"
  >
    <div>
      Edit
    </div>
    <div>
      Delete
    </div>
    <div>
      Close menu
    </div>
  </div>
</DocumentFragment>
`;
